src/test/java/com/imsweb/x12/reader/X12ReaderTest.java

```diff
@@ -72,6 +77,44 @@ public void testSerializeBasic() throws IOException {
         Assert.assertEquals(expected, fromFileUtf8.toX12String(lineBreak).trim());
     }
 
+    /**
+     * Tests the toHtml method that
+     */
+    @Test
+    public void testToHtmlBasic() throws IOException {
+        URL url = this.getClass().getResource("/837_5010/x12_valid.txt");
+
+        X12Reader fromFileUtf8 = new X12Reader(FileType.ANSI837_5010_X222, new File(url.getFile()),
+            StandardCharsets.UTF_8);
+
+        String x12Template = IOUtils.toString(getClass().getResourceAsStream("/html/x12-template.html"), StandardCharsets.UTF_8);
+
+        String x12HtmlSegment = fromFileUtf8.toHtml();
+
+        String fullX12Html = String.format(x12Template, x12HtmlSegment);
+
+        Document doc = Jsoup.parse(fullX12Html);
+        Elements loops = doc.select(".x12-loop");
+        Assert.assertEquals(20, loops.size());
+
+        Elements segments = doc.select(".x12-segment");
+        Assert.assertEquals(38, segments.size());
+
+        Elements elements = doc.select(".x12-element");
+        Assert.assertEquals(216, elements.size());
+
+        // Below is a useful snippet that will create a fully formatted HTML file from x12.
+        // In the src/tests/css directory you will see a "css" directory that contains some css that will format
+        // the X12-html output nicely.
+        File testDir = new File("build", UUID.randomUUID().toString());
+        FileUtils.forceMkdir(testDir);
+        FileUtils.copyDirectory(new File(Paths.get("src", "test", "css").toAbsolutePath().toString()), new File(testDir, "css"));
+        File htmlFileOutput = new File(testDir, "render-x12.html");
+        FileUtils.writeStringToFile(htmlFileOutput, fullX12Html, StandardCharsets.UTF_8);
+        // To view the full HTML document, open the file at htmlFileOutput.getAbsolutePath().
```

Comment:
Sorry for the delay in this review. I must have gotten confused with the XML one and thought everything was closed. These changes look good to me. One change that might be good is moving these lines for creating the HTML file to a lab class instead of having them in the unit test. For example, create a src/test/java/lab directory and add a simple class that shows this example.